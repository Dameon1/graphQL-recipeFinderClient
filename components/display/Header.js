
import React from "react";
import '../styles/header.css';
import { signUserOut } from '../../actions/auth';
import { signingUserOut } from "../../actions/userActions";
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export class Header extends React.Component {

  handleSignOut() {
   this.props.dispatch(signingUserOut());
   this.props.dispatch(signUserOut());
  };

  render() {
    if (!this.props.loggedIn) {
      return (
        <div className='header'>
          <h1 className="headerText">
            what2eat
          </h1>
          <div className="navLinks">
            <Link to="/loginPage" >
              <button className="navLinkButtons">Login</button>
            </Link>
            <Link to="/signUp" >
              <button className="navLinkButtons">Sign up</button>
            </Link>  
            <Link to="/dashboard" >
              <button className="navLinkButtons" >Dashboard</button>
            </Link>
            <Link to="/" >
              <button className="navLinkButtons">Home</button>
            </Link>
          </div>
        </div>
    )}
      return (
        <div className='header'>
          <h1 className="headerText">
            what2eat
          </h1>
          <div className="navLinks">
            <Link to="/dashboard" >
              <button className="navLinkButtons">Dashboard</button>
            </Link>
            <Link to="/myRecipes" >
              <button className="navLinkButtons">My recipes</button>
            </Link> 
            <Link to="/" >
              <button className="navLinkButtons" onClick={ () => { this.handleSignOut() } }>Sign Out</button>
            </Link> 
          </div>
        </div>
    )
  };
};

const mapStateToProps = state => ({
  loggedIn: state.authReducer.loggedIn,
});

export default connect(mapStateToProps)(Header);
