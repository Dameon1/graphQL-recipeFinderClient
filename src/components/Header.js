import React from "react";
import { Link } from '@reach/router';
import User from './User';
import './styles/header.css';
import Spinner from 'react-spinkit';
import SignOut from './SignOut';
// import { signUserOut } from '../../actions/auth';
// import { signingUserOut } from "../../actions/userActions";

 export default class Header extends React.Component {

//   handleSignOut() {
//    this.props.dispatch(signingUserOut());
//    this.props.dispatch(signUserOut());
//   };

//TODO SEPARATE HEADERS INTO SEPARATE FILES
  render() {
    return (      
      <User>
        {(data) => {
          console.log(data.loading);
          if(data.loading) {
           return <Spinner spinnername="circle" fadeIn='none' />;
          }
          if(!data.data.me) {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/signIn" >
                    <button className="navLinkButtons">Login</button>
                  </Link>
                  <Link to="/signUp" >
                    <button className="navLinkButtons">Sign up</button>
                  </Link>  
                  <Link to="/dashboard" >
                    <button className="navLinkButtons" >Dashboard</button>
                  </Link>
                  <Link to="/" >
                    <button className="navLinkButtons">Home</button>
                  </Link>
                </div>
              </div>
            )
          } else {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/dashboard" >
                    <button className="navLinkButtons">Dashboard</button>
                  </Link>
                  <Link to="/myRecipes" >
                    <button className="navLinkButtons">My recipes</button>
                  </Link> 
                  <Link to="/" >
                    <SignOut />
                  </Link> 
                </div>
              </div>
          )
          }
        }}
      </User>
    )

    // if (!this.props.loggedIn) {
    //   return (
        
    // )}
     
  };
};
