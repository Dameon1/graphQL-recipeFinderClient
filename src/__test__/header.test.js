import React from "react";
import { shallow, mount } from "enzyme";
import Header from "../components/Header";
import toJSON from "enzyme-to-json";

describe("<Header />", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});


