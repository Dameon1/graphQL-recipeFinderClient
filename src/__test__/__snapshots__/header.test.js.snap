// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders without crashing 1`] = `
<styled.div>
  <styled.h1>
    what2eat
  </styled.h1>
  <User>
    <Component />
  </User>
</styled.div>
`;
